### PR TITLE
[WIP] - Spring boot starter

### PR DESCRIPTION
**Pull request by gokulchandra**
Started off with a basic spring boot project for #127

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,8 @@
+.DS_Store
```
Comment (rybalkinsd) [inline .gitignore]: `.DS_Store` is a very local thing for mac users. I suggest you to add it to your own `.gitignore` (the one applying for every single project on your laptop)

File: samples/spring/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (rybalkinsd) [inline samples/spring/src/main/resources/application.properties]: Is it possible to remove this file? It looks like heaving a single blank line

File: samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            body {
+                json {
+                    "login" to "user"
+                    "email" to "john.doe@gmail.com"
+                }
+            }
+        }
+        return res.code()
+    }
+}
```
Comment (rybalkinsd) [inline samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt]: Pls, add new line
```suggestion
}

```

File: samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package io.github.rybalkinsd.kohttp.spring_kohttp
+
+import io.github.rybalkinsd.kohttp.dsl.httpGet
+import io.github.rybalkinsd.kohttp.dsl.httpPost
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
```
Comment (rybalkinsd) [inline samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt]: If you're interested in exploring other options of API declaration, I'd suggest to have a look at MVC dsl, https://spring.io/blog/2019/04/03/spring-tips-webmvc-fn-the-functional-dsl-for-spring-mvc

We can have examples with two approaches

File: samples/spring/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation("org.springframework.boot:spring-boot-starter-web")
+	implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
+	implementation("org.jetbrains.kotlin:kotlin-reflect")
+	implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+	implementation(group = "io.github.rybalkinsd", name = "kohttp", version = "0.11.0")
```
Comment (rybalkinsd) [inline samples/spring/build.gradle.kts]: It's better to depend on latest kohttp version here, please check how to do this in android sample :)

**Comment by rybalkinsd** [comment]
I suggest you could create a task for springboot sample, we merge the sample and then iteratively come to our new starter.

How it sounds for you @gokulchandra ?

**Comment by gokulchandra** [comment]
> @gokulchandra I really like how it goes!
> 
> it's a good sample for spring.
> 
> There are several things I want to mention:
> 
>     * we can merge this after a couple of minor fixes
> 
>     * as for proper integration with spring boot we need to create a separate artifact `kohttp-spring`
>       we should enable a way to configure our http client using spring beans like
> 
> 
> ```
> @Bean 
> fun defaultHttpClient() = ...
> ```
> 
> Here is an intro for SB starter https://www.baeldung.com/spring-boot-starters

Ah, yeah that would  be a great addition. I initially assumed its about providing a Spring-boot sample. My Bad! 

As you suggested let me create a task for this and merge it. l can start working on the `spring-starter`. 👍

File: samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt (hunk 4 above)
Comment (gokulchandra) [reply samples/spring/src/main/kotlin/io/github/rybalkinsd/kohttp/spring_kohttp/Controller.kt]: Yep, I can add an example using `Webflux` 👍

**Comment by rybalkinsd** [comment]
> As you suggested let me create a task for this and merge it. l can start working on the `spring-starter`. 👍

Yeah, very welcome to start with this issue. I'm also doing some experiments with this integration at the moment. So if you would like to push any code - it's better to create branch inside the origin repo, to be able to edit it together.

**Pull request merged**